services/wfm/src/main/java/org/openkilda/wfm/share/logger/FlowOperationsDashboardLogger.java

```diff
@@ -0,0 +1,212 @@
+        data.put(FLOW_ID, flowId);
+        data.put(EVENT_TYPE, FLOW_READ_EVENT);
+        proceed(Level.INFO, String.format("Read paths of the flow %s", flowId), data);
+    }
+
+    /**
+     * Log a flow-create event.
+     */
+    public void onFlowCreate(Flow flow) {
+        Map<String, String> data = new HashMap<>();
+        data.put(TAG, "flow-create");
+        data.put(FLOW_ID, flow.getFlowId());
+        data.put(EVENT_TYPE, FLOW_CREATE_EVENT);
+        proceed(Level.INFO, String.format("Create the flow: %s", flow), data);
+    }
+
+    /**
+     * Log a flow-push event.
+     */
+    public void onFlowPush(Flow flow) {
+        Map<String, String> data = new HashMap<>();
+        data.put(TAG, "flow-push");
+        data.put(FLOW_ID, flow.getFlowId());
+        data.put(EVENT_TYPE, FLOW_CREATE_EVENT);
+        proceed(Level.INFO, String.format("Push the flow: %s", flow), data);
+    }
+
+    /**
+     * Log a flow-status-update event.
+     */
+    public void onFlowStatusUpdate(String flowId, FlowStatus status) {
+        Map<String, String> data = new HashMap<>();
+        data.put(TAG, "flow-status-update");
+        data.put(FLOW_ID, flowId);
+        data.put(EVENT_TYPE, STATUS_UPDATE_EVENT);
+        data.put(STATUS, status.toString());
+        proceed(Level.INFO, String.format("Update the status of the flow %s to %s", flowId, status), data);
+    }
+
+    /**
+     * Log a flow-update event.
+     */
+    public void onFlowUpdate(Flow flow) {
+        Map<String, String> data = new HashMap<>();
+        data.put(TAG, "flow-update");
+        data.put(FLOW_ID, flow.getFlowId());
+        data.put(EVENT_TYPE, FLOW_UPDATE_EVENT);
+        proceed(Level.INFO, String.format("Update the flow: %s", flow), data);
+    }
+
+    /**
+     * Log a flow-patch-update event.
+     */
+    public void onFlowPatchUpdate(Flow flow) {
+        Map<String, String> data = new HashMap<>();
+        data.put(TAG, "flow-patch-update");
+        data.put(FLOW_ID, flow.getFlowId());
+        data.put(EVENT_TYPE, FLOW_UPDATE_EVENT);
+        proceed(Level.INFO, String.format("Patch update the flow: %s", flow), data);
+    }
+
+    /**
+     * Log a flow-delete event.
+     */
+    public void onFlowDelete(String flowId) {
+        Map<String, String> data = new HashMap<>();
+        data.put(TAG, "flow-delete");
+        data.put(FLOW_ID, flowId);
+        data.put(EVENT_TYPE, FLOW_DELETE_EVENT);
+        proceed(Level.INFO, String.format("Delete the flow %s", flowId), data);
+    }
+
+    /**
+     * Log a flow-endpoint-swap event.
+     */
+    public void onFlowEndpointSwap(Flow firstFlow, Flow secondFlow) {
+        Map<String, String> data = new HashMap<>();
+        data.put(TAG, "flow-endpoint-swap");
+        data.put(FLOW_ID, firstFlow.getFlowId());
+        data.put(EVENT_TYPE, FLOW_UPDATE_EVENT);
+        proceed(Level.INFO, String.format("Swap end-points for the flows %s / %s", firstFlow, secondFlow), data);
```

Comment:
Do we need whole Flows, or firstFlow.getFlowId() will be enough?

Reply:
Flow is an entity which can be removed, so it can be hard to find information about it after.